src/main/resources/fileTemplates/testMeIncludes/TestMe macros.java.ft

```diff
@@ -55,7 +55,7 @@
 ##
 #macro(renderMockedFields $testedClassFields)
 #foreach($field in $testedClassFields)
-#if($MockitoMockBuilder.isMockable($field))
+#if($hasMocks && $MockitoMockBuilder.isMockable($field))
```

Comment:
1. $hasMocks variable is not defined within this macro file, even if this works for now - it's a bad practice.
2. existing macro renderMockStubs or it's callers do not evaluate hasMocks (they call isMockable transitively). so a stub statement can be generated for field that is not mocked - would result in compilation failure. an integration test which includes this scenario must be added for this change. if you can find the time to do so - that would be great. if not, perhaps I'll be able to do so later..
3. not all templates aligned with new logic

To address these issues:
1. replace all 6 usages of: 
`#set($hasMocks=$MockitoMockBuilder.hasMockable($TESTED_CLASS.fields)`
with 
`#set($hasMocks=$MockitoMockBuilder.hasMocks($TESTED_CLASS)`
2. Add a method `MockitoMockBuilder.hasMocks(Type)` that will
evaluate `hasMockable(fields) && canMockViaCtors(type)`.
make hasMockable private
3. replace all 7 usages of 
`#if($MockitoMockBuilder.shouldStub(`
with 
`#if($hasMocks && $MockitoMockBuilder.shouldStub(`